radio_beam/utils.py

```diff
@@ -16,6 +19,99 @@ class RadioBeamDeprecationWarning(Warning):
     pass
 
 
+def deconvolve_opt(beamprops1, beamprops2, failure_returns_pointlike=False):
+    """
+    An optimized, non-Quantity version of beam deconvolution.
```

Comment:
is it worth factoring out the old `deconvolve` and having it use this to minimize code duplication, or is that not worth it?

Reply:
Good point. `deconvolve` now wraps `deconvolve_optimized`